Continuous okapi registration

sadatay opened the pull request:
This PR contains two chunks of work:

1. _Point the CI/CD config towards `okapi-demo` cluster
This simply adjusts the configuration to point at the `okapi-demo` k8s cluster instead of the cluster we used to test this work in the early stages.

2. Automatically register module to Okapi during deployment
This alters the `deploy.sh` script to include a workflow for re-registering the module to Okapi upon successful merge to master.  The workflow uses `curl` to
a) de-select any existing version of the module from the tenant 
b) delete existing discovery records for the module 
c) delete existing registrations for the module
d) register the new version of the module
e) discover the new version of the module
f) link the new version of the module to the tenant

sadatay commented:
we should probably yank all the commented-out stuff before merge ✨

cowboyd commented:
We should also probably change the name of the repo to:

`folio-rm-api-ebsco`

How onerous of a change is that likely to be?

cowboyd commented:
or `mod-resources-ebsco`

sadatay commented on deploy.sh:
```diff
@@ -0,0 +1,39 @@
+
+bash_escape() ( printf '\\033[%dm' $1; );
+RESET=$(bash_escape 0); BLUE=$(bash_escape 34);
+put_info() ( printf "${BLUE}[INFO]${RESET} $1\n");
```
using these colorized helpers is a bit of a gamble, since we haven't tested it.  if anybody is uncomfortable i can back it out. the `-x` flag gives us what we need here, it's just not particularly ergonomic

cowboyd commented:
Also, now that we know that the role of this module is to serve as an adapter between a Folio deployment and an Ebsco knowledge base.

cowboyd commented on README.md:
```diff
@@ -7,3 +9,15 @@ Copyright (C) 2017 [The Open Library Foundation][1]
+
+## Additional information
+
+The [raml-module-builder](https://github.com/folio-org/raml-module-builder) framework.
```
I don't think we're going to be using the raml-module-builder since this isn't a Java application.

cherewaty commented on .travis.yml:
```diff
@@ -0,0 +1,50 @@
+    - PROJECT_ID=okapi-173322
+    - CLUSTER_NAME=okapi-mod-resource--management-pipeline
+    - CLOUDSDK_COMPUTE_ZONE=us-central1-a
+    - DOCKER_IMAGE_NAME=thefrontside/folio-mod-resource-management
```
We should also change the docker hub image name to reflect the new repo name.

sadatay replied:
nice catch!  i knew i was forgetting something

Commit:
Point the CI/CD config towards the `okapi-demo` cluster

When we were testing this automated deployment with DockerHub and GCE we used an isolated cluster.  Now that we need to tie in Okapi registration, we should point it towards our okapi-demo cluster.

Commit:
Automatically register module to Okapi during deployment

ROUGH DRAFT
